File: spaceInvaders/src/main/java/spaceInvaders/Disparo.java
package spaceInvaders;

import javafx.scene.image.Image;
import javafx.scene.image.ImageView;

public class Disparo {

    private static final String SPRITE_PATH = "/sprites/disparo.png"; 
    private final ImageView spriteView;
    private double posX;
    private double posY;
    private final double velocidade; 
    private final boolean isPlayerDisparo;

    public Disparo(double startX, double startY, boolean isPlayer) {
        this.posX = startX;
        this.posY = startY;
        this.isPlayerDisparo = isPlayer;
        this.velocidade = isPlayer ? -5 : 3; // player dispara pra cima, alien pra baixo

        this.spriteView = new ImageView(new Image(SPRITE_PATH));
        this.spriteView.setFitWidth(10);
        this.spriteView.setFitHeight(20);
        this.spriteView.setX(posX);
        this.spriteView.setY(posY);
    }

    public void mover() {
        posY += velocidade;
        spriteView.setY(posY);
    }

    // Getters
    public ImageView getSpriteView() {
        return spriteView;
    }

    public double getPosX() {
        return posX;
    }

    public double getPosY() {
        return posY;
    }

    public boolean isPlayerDisparo() {
        return isPlayerDisparo;
    }
}

File: SpaceInvaders(Python)/src/Disparo.py
# -*- coding: utf-8 -*-
from PySide6.QtGui import QPixmap, Qt
from PySide6.QtWidgets import QLabel
from PySide6.QtCore import QSize


class Disparo:
    SPRITE_PATH = "sprites/disparo.png"

    def __init__(self, x: int, y: int, is_player: bool):
        self._pos_x = x
        self._pos_y = y
        self._is_player = is_player
        self._velocidade = -5 if is_player else 3

        sprite = QPixmap(self.SPRITE_PATH).scaled(QSize(10, 20))
        self._sprite_view = QLabel()
        self._sprite_view.setAttribute(Qt.WA_TranslucentBackground)
        self._sprite_view.setPixmap(sprite)
        self._sprite_view.setGeometry(x, y, 10, 20)

    def mover(self):
        self._pos_y += self._velocidade
        self._sprite_view.move(int(self._pos_x), int(self._pos_y))

    def get_sprite_view(self):
        return self._sprite_view

    def get_pos_x(self):
        return self._pos_x

    def get_pos_y(self):
        return self._pos_y

    def is_player_disparo(self):
        return self._is_player